### ISSUE TITLE
Explore dropping Pandas requirement

### ISSUE THREAD

**Issue by ndrezn** [plotly/plotly.py#4834]
In https://github.com/plotly/plotly.py/pull/4790, nearly all Pandas functionality is removed from Plotly.py, _except_ some of our trendine functions.

Narwhals does not yet support `rolling`, `expanding` nor `ewm`, and until that support is added, we need to use Pandas to calculate trendlines. 

Those functions _are_ on the Narwhals roadmap, and we should have a quick follow-up in Plotly.py to implement those changes after the first set of Narwhals changes are made.

See: https://github.com/plotly/plotly.py/pull/4790/files/6676061152527160b44dbf67ecf6b89e46eef3b2#diff-e667a84629ee8d2879cf22a88c8ff80bcfec1c35be69ab9f50be5938c7089ddaR114-R129

**Comment by FBruzzesi** [comment]
Tiny update on this topic: after merging all the cumulative features supported from polars into the narwhals api, I am able to replicate almost all the expanding functions that pandas supports. The following 3 are sadly left out: `median`, `quantile`, and `rank`. I don't really see a way to implement them without a loop for polars and pyarrow considering their api's.

**Comment by emilykl** [comment]
Hi @FBruzzesi! I just stumbled across this issue again and I'm curious whether there's any change on the Narwhals side in terms of being able to implement `median`, `quantile`, and `rank`.

Based on your last update I'm sort of assuming the answer is "no" but I figured I'd ask just in case!

**Comment by FBruzzesi** [comment]
Hey @emilykl 👋🏼 long time no ~~see~~ talk! I must be honest, we have an issue still open which we didn't prioritize.

For the cumulative remaining features, IIRC the blocker was the polars API not supporting them, and we mirror polars API. For ewm and rolling we are _not_ feature complete/matching polars.

We currently support:

* `cum_{sum,count,min,max,prod}` all eager backends
* `rolling_{sum,mean,var,std}` all eager backends
* `ewm_mean` for pandas and polars only, pyarrow is still not implemented

I can't recall all the operations that pandas and plotly.express support with these variants, but I do remember being able to implement most of them derived from the above

**Comment by emilykl** [comment]
Thanks @FBruzzesi! This isn't an urgent priority but it _would_ be cool if we could drop Pandas as a Plotly Express requirement. Hopefully we'll have bandwidth to look into it further at some point.

**Comment by FBruzzesi** [comment]
I think it would be nice if you could map all the functionalities you would like to see before translating to narwhals.

The other decision point is which behavior you want to follow: it's very likely that pandas and polars disagree on some operations, and 
as you know, in narwhals we would align every backend with polars behavior.

I suppose you don't want any breaking change for users, and that would mean:

1. Implement missing functionalities in narwhals
2. Use those functionalities in plotly to align with pandas behavior

**Comment by emilykl** [comment]
@FBruzzesi I agree, as a first step we would need to make a list of exactly what functionality is missing in Narwhals, if any. 

You raise a good point as well about behavior differences. We do try very hard to avoid breaking changes. There are two exceptions though:

1. Minor behavior differences where the new behavior is clearly better; in this case we can consider it a bug fix and release in a minor version bump
2. Major behavior differences which are not-worse than previous behavior; these are OK if we release them in a major version, consider the consequences, and have a good justification

Ultimately due to the complexity of some of the data transformations in plotly.js we'd lean pretty heavily on our testing suite to identify differences, and make decisions on a case-by-case basis. I think it makes sense to consider this transition as a potential project for the next major plotly.py release, which would give us the flexibility to allow some breaking changes if we think they're justified.
